### PR TITLE
feat(`evm`/`cheatcodes`): Use `alloy-signer`, rm `ethers-signers`/`ethers-core`

### PR DESCRIPTION
**Pull request by Evalir**
## Motivation

We were still using ethers for signers on cheatcodes.

## Solution

Replaces ethers-signers with alloy signers, and removes the now unused ethers deps from `cheatcodes`.

This isolates signers on `forge script`/`cast`, in which we're still converting to ethers signers.

### CODE REVIEW

File: crates/cheatcodes/src/utils.rs [hunk 1]
```diff
@@ -155,21 +154,20 @@ fn create_wallet(private_key: &U256, label: Option<&str>, state: &mut Cheatcodes
...
-    let wallet = parse_wallet(private_key)?.with_chain_id(chain_id);
+pub(super) fn sign(private_key: &U256, digest: &B256) -> Result {
+    // The `ecrecover` precompile does not use EIP-155. No chain ID is needed.
+    let wallet = parse_wallet(private_key)?;
 
-    // The `ecrecover` precompile does not use EIP-155
-    let sig = wallet.sign_hash(digest.to_ethers())?;
-    let recovered = sig.recover(digest.to_ethers())?.to_alloy();
+    let sig = wallet.sign_hash_sync(*digest)?;
+    let recovered = sig.recover_address_from_prehash(digest)?;
```
Comment (mattsse) [inline crates/cheatcodes/src/utils.rs:162]: so the previous impl also didn't require the chain id?

**Review by mattsse — APPROVED** [review]
pending @DaniPopes

File: crates/cheatcodes/src/utils.rs (hunk 1 above)
Comment (Evalir) [reply crates/cheatcodes/src/utils.rs:162]: correct

File: Cargo.toml [hunk 2]
```diff
@@ -188,6 +188,8 @@ toml = "0.8"
 tracing = "0.1"
 tracing-subscriber = "0.3"
 evm-disassembler = "0.4"
+# needed as documented in https://github.com/foundry-rs/foundry/pull/6358
```
Comment (DaniPopes) [inline Cargo.toml:191]:
```suggestion
```

File: Cargo.lock [hunk 3]
```diff
@@ -2978,11 +2979,10 @@ dependencies = [
  "alloy-primitives",
  "alloy-providers",
  "alloy-rpc-types",
+ "alloy-signer",
  "alloy-sol-types",
  "base64 0.21.7",
  "const-hex",
- "ethers-core",
- "ethers-signers",
```
Comment (DaniPopes) [inline Cargo.lock:2985]: very nice

**Review by DaniPopes — APPROVED** [review]

File: Cargo.lock (hunk 3 above)
Comment (Evalir) [reply Cargo.lock:2985]: simply lovely

**Pull request merged**